test/integration/helpers/createCompiler.ts

```diff
@@ -0,0 +1,162 @@
+import webpack from 'webpack';
+import VueLoaderPlugin from 'vue-loader/lib/plugin';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as copyDir from 'copy-dir';
+import * as rimraf from 'rimraf';
+import { ForkTsCheckerWebpackPlugin, webpackMajorVersion } from './';
+
+if (!beforeAll || !afterAll) {
+  throw new Error('file should not be included outside of jest!');
+}
+
+const baseTmpDir = path.resolve(__dirname, '../../tmp');
+let tmpDirParent: string;
+let lastInstantiatedPlugin: ForkTsCheckerWebpackPlugin | undefined;
+
+beforeAll(function init() {
+  if (!fs.existsSync(baseTmpDir)) fs.mkdirSync(baseTmpDir);
+
+  tmpDirParent = fs.mkdtempSync(
+    path.join(baseTmpDir, 'fork-ts-checker-webpack-plugin-test')
+  );
+});
+
+afterAll(function cleanUp() {
+  rimraf.sync(tmpDirParent);
+});
+
+afterEach(function killPlugin() {
+  if (lastInstantiatedPlugin && lastInstantiatedPlugin['service']) {
+    lastInstantiatedPlugin['killService']();
+  }
+});
+
+export interface CreateCompilerOptions {
+  entryPoint: string;
+  context: string;
+  happyPackMode: boolean;
+  pluginOptions: Partial<ForkTsCheckerWebpackPlugin.Options>;
+  prepareWebpackConfig(config: webpack.Configuration): webpack.Configuration;
+  nodeRequires: string[];
+  normalizePaths: boolean;
+}
+
+const defaultOptions: Partial<ForkTsCheckerWebpackPlugin.Options> = {
+  silent: true
+};
+
+type CreateCompilerResults = {
+  compiler: webpack.Compiler;
+  readonly compilerConfig: webpack.Configuration;
+  plugin: ForkTsCheckerWebpackPlugin;
+  contextDir: string;
+  outDir: string;
+  tmpDir: string;
+};
+
+function prepareDirectory({ context }: { context: string }) {
+  if (!fs.existsSync(tmpDirParent)) fs.mkdirSync(tmpDirParent);
+  const tmpDir = fs.mkdtempSync(path.join(tmpDirParent, 'test'));
+  const contextDir = path.resolve(tmpDir, context);
+  const outDir = path.resolve(tmpDir, 'out');
+  fs.mkdirSync(contextDir);
+  fs.mkdirSync(outDir);
+  copyDir.sync(path.resolve(__dirname, '..', context), contextDir);
+  return { contextDir, outDir, tmpDir };
+}
+
+function doNormalizePaths(
```

Comment:
Why the name is **do**NormalizePaths? Do we need `do` here? If it is because of names conflict in the function below, I think it would be better to alias `normalizePaths` variable inside the `createCompiler` function.

Reply:
As it gets a list of diagnostics and returns a list of diagnostics, maybe it would be better to change the name to `normalizeDiagnosticsPaths`